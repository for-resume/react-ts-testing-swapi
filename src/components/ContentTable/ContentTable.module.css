.table {
  margin-top: 2rem;
  width: 100%;
  padding: 0 2rem;
  text-align: center;
  border-collapse: collapse;
}

.table td,
.table th {
  border: 2px solid #222;
  padding: 0.8rem 2rem;
  color: #222;
}

.table th {
  background-color: #222;
  color: #fff;
}

.table td:first-child {
  font-weight: bold;
}
