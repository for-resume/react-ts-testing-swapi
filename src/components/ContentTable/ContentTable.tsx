import React from 'react';

import { Character } from '../../App';

import classes from './ContentTable.module.css';

type ContentTableProps = {
  people: Array<Character> | [];
};

const ContentTable: React.FC<ContentTableProps> = ({
  people,
}: ContentTableProps) => (
  <table className={classes.table}>
    <tbody>
      <tr>
        <th>Имя</th>
        <th>Пол</th>
        <th>Вес</th>
        <th>Цвет глаз</th>
      </tr>

      {people.length ? (
        people.map(({ name, gender, mass, eye_color }: Character) => (
          <tr key={name}>
            <td>{name}</td>
            <td>{gender}</td>
            <td>{mass}</td>
            <td>{eye_color}</td>
          </tr>
        ))
      ) : (
        <tr>
          <td colSpan={4}>Нет данных</td>
        </tr>
      )}
    </tbody>
  </table>
);

export default ContentTable;
