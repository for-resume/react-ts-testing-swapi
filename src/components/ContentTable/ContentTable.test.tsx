import React from 'react';
import { render } from '@testing-library/react';

import ContentTable from './ContentTable';

describe('ContentTable', () => {
  it('matches snapshot if people not empty', () => {
    const { asFragment } = render(
      <ContentTable
        people={[
          {
            name: 'Luke Skywalker',
            gender: 'male',
            mass: '77',
            eye_color: 'blue',
          },
          {
            name: 'C-3PO',
            gender: 'n/a',
            mass: '75',
            eye_color: 'yellow',
          },
        ]}
      />,
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('matches snapshot if people - empty []', () => {
    const { asFragment } = render(<ContentTable people={[]} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
