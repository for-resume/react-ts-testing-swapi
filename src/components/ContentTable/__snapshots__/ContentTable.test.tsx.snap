// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ContentTable matches snapshot if people - empty [] 1`] = `
<DocumentFragment>
  <table
    class="table"
  >
    <tbody>
      <tr>
        <th>
          Имя
        </th>
        <th>
          Пол
        </th>
        <th>
          Вес
        </th>
        <th>
          Цвет глаз
        </th>
      </tr>
      <tr>
        <td
          colspan="4"
        >
          Нет данных
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;

exports[`ContentTable matches snapshot if people not empty 1`] = `
<DocumentFragment>
  <table
    class="table"
  >
    <tbody>
      <tr>
        <th>
          Имя
        </th>
        <th>
          Пол
        </th>
        <th>
          Вес
        </th>
        <th>
          Цвет глаз
        </th>
      </tr>
      <tr>
        <td>
          Luke Skywalker
        </td>
        <td>
          male
        </td>
        <td>
          77
        </td>
        <td>
          blue
        </td>
      </tr>
      <tr>
        <td>
          C-3PO
        </td>
        <td>
          n/a
        </td>
        <td>
          75
        </td>
        <td>
          yellow
        </td>
      </tr>
    </tbody>
  </table>
</DocumentFragment>
`;
